crates/bevy_asset/src/asset_server.rs

```diff
@@ -93,6 +93,15 @@ impl AssetServer {
     }
 
     pub(crate) fn register_asset_type<T: Asset>(&self) -> Assets<T> {
+        if self
```

Comment:
rather than grab a read lock and then a write lock immediately after, I think we should just grab a single write lock and reuse it across both cases (especially given that the "read only" case is the "unexpected" / "uncommon" case).